app/models/target.rb

```diff
@@ -18,9 +18,21 @@
 #  index_targets_on_user_id   (user_id)
 #
 class Target < ApplicationRecord
+  USER_MAX_TARGET_LIMIT = 10
+
   belongs_to :user
   belongs_to :topic
 
   validates :title, :latitude, :longitude, :radius, presence: true
   validates :radius, numericality: { greater_than: 0 }
+
+  validate :target_limit, on: :create
+
+  private
+
+  def target_limit
+    return unless user.targets.count == USER_MAX_TARGET_LIMIT
```

Comment:
Take a look at this: https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache